app/assets/javascripts/admin/products/services/product_image_service.js.coffee

```diff
@@ -14,4 +14,8 @@ angular.module("ofn.admin").factory "ProductImageService", (FileUploader, SpreeA
         product.thumb_url = response.thumb_url
         product.image_url = response.image_url
       @imageUploader.onErrorItem = (image, response) =>
-        alert(response.errors.join("\n"))
+        if Array.isArray(response.errors)
+          message = response.errors.join("\n")
+        else
+          message = response.error.toString()
+        alert(message)
```

Comment:
I think it would be better to change the Ruby code to always send an array. It's called error**s** after all. But this is a good solution in the short term.

Reply:
Actually, it's called `error` (without the `s`) : 
```json
{"error":"You are not authorized to perform that action."}
```

Reply:
Oh dear, you are so right! I was thinking of the JSON:API standard which, I think, recommends using an array.